### PR TITLE
Adding vuid 04605

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

File: layers/descriptor_sets.cpp [hunk 1]
```diff
@@ -429,6 +429,13 @@ bool cvdescriptorset::ValidateDescriptorSetLayoutCreateInfo(
             }
         }
 
+        if (binding_info.descriptorType == VK_DESCRIPTOR_TYPE_MUTABLE_VALVE && binding_info.pImmutableSamplers != NULL) {
```
Comment (ncesario-lunarg) [inline layers/descriptor_sets.cpp:432]: Would be nice to stick with `nullptr` over `NULL` for C++ code base IMHO.

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 2]
```diff
@@ -14014,3 +14014,29 @@ TEST_F(VkLayerTest, SparseImageMemoryBindOffset) {
...
+
+    ASSERT_NO_FATAL_FAILURE(Init());
+
+    VkSampler sampler = VK_NULL_HANDLE;
+    VkSamplerCreateInfo sampler_info = SafeSaneSamplerCreateInfo();
+    vk::CreateSampler(m_device->device(), &sampler_info, NULL, &sampler);
+
+    VkDescriptorSetLayoutBinding binding = {};
+    binding.binding = 0;
+    binding.descriptorType = VK_DESCRIPTOR_TYPE_MUTABLE_VALVE;
```
Comment (ncesario-lunarg) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:14029]: Do we need to check if this extension is available, or maybe that already happened somewhere else?

**Review by ncesario-lunarg — COMMENTED** [review]
Couple minor nits/questions, but LGTM otherwise!

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 2 above)
Comment (ziga-lunarg) [reply tests/vklayertests_buffer_image_memory_sampler.cpp:14147]: Added

**Pull request merged**